File: android_java/demo_menu/app/src/main/java/com/heyangpeng/demo_menu/MainActivity.java
package com.heyangpeng.demo_menu;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.ViewGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.add_item:
                startActivity(new Intent().setClass(this,SecondActivity.class));
                Toast.makeText(this, "START_SECOND", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.remove_item:
                Toast.makeText(this, "REMOVE", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.group1:
                Toast.makeText(this, "group1", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.group2:
                Toast.makeText(this, "group2", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.group3:
                Toast.makeText(this, "group3", Toast.LENGTH_SHORT).show();
                return true;
            default:
                return super.onOptionsItemSelected(item);

        }
    }

}

File: android_kotlin/menu_demo/app/src/main/java/com/heyangpeng/menu_demo/MainActivity.kt
package com.heyangpeng.menu_demo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast

open class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.add_item -> {
                startActivity(Intent(this, SecondActivity::class.java))
                Toast.makeText(this, "START_SECOND", Toast.LENGTH_SHORT).show()
            }
            R.id.remove_item -> Toast.makeText(this, "REMOVE_ITEM", Toast.LENGTH_SHORT).show()
            R.id.group1 -> Toast.makeText(this, "GROUP1", Toast.LENGTH_SHORT).show()
            R.id.group2 -> Toast.makeText(this, "GROUP2", Toast.LENGTH_SHORT).show()
            R.id.group3 -> Toast.makeText(this, "GROUP3", Toast.LENGTH_SHORT).show()
        }
        return super.onOptionsItemSelected(item)
    }
}